BIDSTools/Process_BIDS_DATA.py

```diff
@@ -0,0 +1,519 @@
+                           f.endswith(".tsv")]
+
+    tsv_json_files_list = [f for f in tsv_json_files_list if
+                           os.path.isfile(os.path.join(output_dir, f))]
+
+    script_dir = os.path.dirname(__file__)
+    agnostic_template_dir = os.path.join(script_dir, 'template_agnotic_file')
+
+    list_template_files = os.listdir(agnostic_template_dir)
+    list_template_files = [f for f in list_template_files if
+                           os.path.isfile(os.path.join(agnostic_template_dir, f)) and f.endswith(
+                               ".json")]
+
+    for template_name in list_template_files:
+        template_base_name, template_ext = os.path.splitext(template_name)
+
+        template_path = os.path.join(agnostic_template_dir, template_name)
+        with open(template_path, 'r') as template_file:
+            template_content = json.load(template_file)
+
+            if template_content:
+                for file_name in tsv_json_files_list:
+                    if file_name.startswith(template_base_name):
+                        writeheader(template_content, file_name, output_dir)
+
+
+def writeheader(template_content, file_name, output_dir):
+    """
+    Extract headers from a template file and write them to a corresponding TSV file.
+
+    Parameters
+    ----------
+    template_content : dict
+        The JSON data from the template file.
+    file_name : str
+        Name of the corresponding TSV file for this template.
+    output_dir : str
+        Path where the corresponding TSV file is located.
+
+    Returns
+    -------
+    None
+        The TSV file is updated with the corresponding header.
+    """
+    file_path = os.path.join(output_dir, file_name)
+
+    # Extract header from template content
+    header = '\t'.join(template_content.keys()) + '\n'
+
+    # Open the file for reading and writing
+    with open(file_path, 'r+') as f:
+        # Read the existing content of the file
+        existing_content = f.read()
+
+        # Move the cursor to the beginning of the file
+        f.seek(0, 0)
+
+        # Write the new header and then the existing content
+        f.write(header + existing_content)
+
+
+def bids_dataset_processed(output_dir, experiment):
+    """
+    Creates all necessary directories for each experiment (each row in the metadata file).
+
+    Parameters
+    ----------
+    output_dir : str
+        Path to the base directory where BIDS data should be saved.
+    experiment : Experiment
+        An object from the Experiment class representing a row of the metadata file.
+
+    Returns
+    -------
+    tuple A tuple containing the final directory path created for the experiment
+    and a list of experiments already processed.
+
+    Description
+    -----------
+    This function processes an experiment by creating the necessary directories based on its attributes.
+    It creates a subject directory, and if the data type requires a session directory, it creates that as well.
```

Comment:
I think it does not create a subject directory in all cases, but only if it is necessary for the input experiment... please correct the text! (same for session and etc.)